pkg/cmd/cluster/connect/connect.go

```diff
@@ -38,12 +38,12 @@ type options struct {
 
 func NewConnectCommand(f *factory.Factory) *cobra.Command {
 	opts := &options{
-		Config:     f.Config,
-		Connection: f.Connection,
-		Logger:     f.Logger,
-		IO:         f.IOStreams,
-		localizer:  f.Localizer,
-		Context:    f.Context,
+		Connection:     f.Connection,
+		Logger:         f.Logger,
+		IO:             f.IOStreams,
+		localizer:      f.Localizer,
+		Context:        f.Context,
+		ServiceContext: f.ServiceContext,
```

Comment:
This pattern will be really restrictive as service context needs to be fetched before command is being build. 
That drives number of limitations to our command line.
Let's discuss